doc/rados/troubleshooting/troubleshooting-mon.rst

```diff
@@ -383,6 +383,80 @@ example::
 
 	iptables -A INPUT -m multiport -p tcp -s {ip-address}/{netmask} --dports 6789,6800:7300 -j ACCEPT
 
+Monitor Store Failures
+======================
+
+Symptoms of store corruption
+----------------------------
+
+Ceph monitor stores the `cluster map`_ in a key/value store. If a monitor in a
+cluster fails due to the key/value store corruption, following error messages
+might be found in the monitor log::
+
+  Corruption: error in middle of record
+
+or::
+
+  Corruption: 1 missing files; e.g.: /var/lib/ceph/mon/mon.0/store.db/1234567.ldb
+
+Recovery using healthy monitor(s)
+---------------------------------
+
+If there is any survivers, we can always `replace`_ the corrupted one with a
+new one. And after booting up, the new joiner will sync up with a healthy
+peer, and once it is fully sync'ed, it will be able to serve the clients.
+
+Recovery using OSDs
+-------------------
+
+But what if all monitors fails at the same time? Since users are encouraged to
+deploy at least three monitors in a Ceph cluster, the chance of simultaneous
+failure is rare. But unplanned power-downs in a data centre could fail the
+underlying filesystem, and hence kill all the monitors. In this case, we can
+recover the monitor store with the information stored in OSDs.::
+
+  ms=/tmp/mon-store
+  mkdir $ms
+  # collect the cluster map from OSDs
+  for host in $hosts; do
+    rsync -avz /tmp/mon-store user@host:$ms
```

Comment:
As mentioned to @tchaikov on IRC, it bothers me that the admin has to copy the osd data from each osd to the node where recovery is happening. This seems slow and unnecessary. We could simply grab the maps from `current/meta/` from all osds.

And, even though this is not something we expect to happen often (ideally, at all), I can imagine how an admin would be closing to tears if he had to copy every. single. osd. to get to the maps during DR. Poor admin. :(